python3.5 deprecated API warnings

Lots of deprecated API warnings when building for python 3.5

msbt_msbt.c(170): warning C4996: 'WSAStringToAddressA': Use WSAStringToAddressW() instead or define _WINSOCK_DEPRECATED
_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3623): note: see declaration of 'WSAStringToAddressA'
msbt_msbt.c(241): warning C4996: 'WSAStringToAddressA': Use WSAStringToAddressW() instead or define _WINSOCK_DEPRECATED
_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3623): note: see declaration of 'WSAStringToAddressA'
msbt_msbt.c(358): warning C4996: 'WSADuplicateSocketA': Use WSADuplicateSocketW() instead or define _WINSOCK_DEPRECATED
_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(2872): note: see declaration of 'WSADuplicateSocketA'
msbt_msbt.c(365): warning C4996: 'WSASocketA': Use WSASocketW() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to di
sable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3457): note: see declaration of 'WSASocketA'
msbt_msbt.c(479): warning C4996: 'WSAStringToAddressA': Use WSAStringToAddressW() instead or define _WINSOCK_DEPRECATED
_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3623): note: see declaration of 'WSAStringToAddressA'
msbt_msbt.c(585): warning C4996: 'WSALookupServiceBeginA': Use WSALookupServiceBeginW() instead or define _WINSOCK_DEPR
ECATED_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3693): note: see declaration of 'WSALookupServiceBeginA'
msbt_msbt.c(607): warning C4996: 'WSALookupServiceNextA': Use WSALookupServiceNextW() instead or define _WINSOCK_DEPREC
ATED_NO_WARNINGS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(3752): note: see declaration of 'WSALookupServiceNextA'
msbt_msbt.c(720): warning C4996: 'WSASetServiceA': Use WSASetServiceW() instead or define _WINSOCK_DEPRECATED_NO_WARNIN
GS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(4157): note: see declaration of 'WSASetServiceA'
msbt_msbt.c(775): warning C4996: 'WSASetServiceA': Use WSASetServiceW() instead or define _WINSOCK_DEPRECATED_NO_WARNIN
GS to disable deprecated API warnings
C:\Program Files (x86)\Windows Kits\8.1\include\um\winsock2.h(4157): note: see declaration of 'WSASetServiceA'
## 


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26879342-python3-5-deprecated-api-warnings?utm_campaign=plugin&utm_content=tracker%2F13568755&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F13568755&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/msbt/_msbt.c
+++ b/msbt/_msbt.c
@@ -1,3 +1,7 @@
+#ifndef UNICODE
+#define UNICODE
+#endif
+
 #include <winsock2.h>
 #include <ws2bth.h>
 #include <BluetoothAPIs.h>
@@ -150,7 +154,7 @@ PyDoc_STRVAR(msbt_socket_doc, "TODO");
 static PyObject *
 msbt_bind(PyObject *self, PyObject *args)
 {
-    char *addrstr = NULL;
+    wchar_t *addrstr = NULL;
     int addrstrlen = -1;
     int sockfd = -1;
     int port = -1;
@@ -230,7 +234,7 @@ static PyObject *
 msbt_connect(PyObject *self, PyObject *args)
 {
     int sockfd = -1;
-    char *addrstr = NULL;
+    wchar_t *addrstr = NULL;
     int port = -1;
     SOCKADDR_BTH sa = { 0 };
     int sa_len = sizeof(sa);
@@ -543,7 +547,7 @@ msbt_lookup_name(PyObject *self, PyObject *args)
     BLUETOOTH_FIND_RADIO_PARAMS p = { sizeof(p) };
     HBLUETOOTH_RADIO_FIND fhandle = NULL;
     BLUETOOTH_DEVICE_INFO dinfo = { 0 };
-    char *addrstr = NULL;
+    wchar_t *addrstr = NULL;
     SOCKADDR_BTH sa = { 0 };
     int sa_len = sizeof(sa);
     DWORD status;
@@ -599,7 +603,7 @@ msbt_find_service(PyObject *self, PyObject *args)
 	qs->dwSize = sizeof(WSAQUERYSET);
 	qs->dwNameSpace = NS_BTH;
     qs->dwNumberOfCsAddrs = 0;
-    qs->lpszContext = (LPSTR) localAddressBuf;
+    qs->lpszContext = (LPWSTR) localAddressBuf;
 
     if( 0 == strcmp( addrstr, "localhost" ) ) {
         // find the Bluetooth address of the first local adapter. 
@@ -692,10 +696,10 @@ msbt_find_service(PyObject *self, PyObject *args)
             dict_set_strings( record, "host", localAddressBuf );
             
             // set service name
-            dict_set_strings( record, "name", qs->lpszServiceInstanceName );
+            dict_set_strings( record, "name", (const char*) qs->lpszServiceInstanceName );
 
             // set description
-            dict_set_strings( record, "description", qs->lpszComment );
+            dict_set_strings( record, "description", (const char*) qs->lpszComment );
 
             // set protocol and port
             csinfo = qs->lpcsaBuffer;
@@ -815,8 +819,8 @@ msbt_set_service(PyObject *self, PyObject *args)
 
 	SOCKADDR_BTH sa = { 0 };
 	int sa_len = sizeof(sa);
-    char *service_name = NULL;
-    char *service_desc = NULL;
+    wchar_t *service_name = NULL;
+    wchar_t *service_desc = NULL;
     char *service_class_id_str = NULL;
 	CSADDR_INFO sockInfo = { 0 };
     GUID uuid = { 0 };

